src/ArmTemplates/Common/Constants/GlobalConstants.cs

```diff
@@ -18,7 +18,7 @@ public static class GlobalConstants
         public const string azAccessToken = "account get-access-token --query \"accessToken\" --output json";
         public const string azSubscriptionId = "account show --query id -o json";
 
-        public const string BaseManagementAzureUrl = "https://management.azure.com";
+        public static string BaseManagementAzureUrl = "https://management.azure.com";
```

Comment:
I don't really mind, but it's not a constant anymore (and it's still in a GlobalConstants file). Maybe there's a better place to put this? What about the immutable extractor settings that are passed around everywhere?

Reply:
This is just for letting people still set it. We can not change const in runtime with normal code.

Believe me, I tried to make an `ApiSettings` class and pass it all over the place (when I was fixing the Arm.ApiVersion bug), but that PR was very big in short amount of time. I didn't finish it because of that (also because Creator is affected there, and I have to either write some real bad hardcode or to leave that `GlobalConstants`. Both approaches are bad for me). I think that PR would be 150+ files affected really -_-

So I thought we can do it later when we are finished with `Creator` refactoring. Then we can make all parameters and settings reach any area of code using DI easily and change that.

What do you think?

Reply:
Fair enough. I don't mind leaving it here for now if it requires a massive rewrite.